### PR TITLE
Make ParameterEstimator thread-safe

### LINKED ISSUE #14010: ParameterEstimator is not thread safe
While adding some unit tests for MuonCalculateAsymmetry (issue #13952) I realized that ParameterEstimator is not thread safe.

### PR DESCRIPTION
**Pull request by raquelalvarezbanos**
Fixes #14010 

Code review should be enough. As this is a purely internal issue, there is no need to update the release notes.

### CODE REVIEW

File: Framework/CurveFitting/src/ParameterEstimator.cpp [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 #include "MantidKernel/Logger.h"
 
+#include <Poco/Timer.h>
```
Comment (martyngigg) [inline Framework/CurveFitting/src/ParameterEstimator.cpp]: Could we change this to `#include <Poco/Mutex.h>`, we don't need the Timer I don't think?

File: Framework/CurveFitting/src/ParameterEstimator.cpp [hunk 2]
```diff
@@ -18,20 +20,42 @@ using namespace Functions;
 /// The logger.
 Kernel::Logger g_log("ParameterEstimator");
 
+/// Mutex to prevent simultaneous access to functionMap
+static Poco::Mutex mutex;
+
```
Comment (martyngigg) [inline Framework/CurveFitting/src/ParameterEstimator.cpp]: A more C++ way to define a variable just for a single `.cpp` file would be
```c++
namespace {
  Poco::Mutex FUNCTION_MAP_MUTEX;
}
This means the name is not visible at all outside of the file whereas `static` could be still found.

**Comment by martyngigg** [comment]
Other than the minor style points above this is good and the correct thing to do.

**Comment by raquelalvarezbanos** [comment]
Thanks Martyn, I've added those changes.

**Pull request merged**